packages/react-core/src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Icon/icon';
+import { css } from '@patternfly/react-styles';
+import { Spinner } from '../Spinner';
+
+export interface IconComponentProps extends Omit<React.HTMLProps<HTMLSpanElement>, 'size'> {
+  /** Icon content when isInProgress is set to false */
+  children?: React.ReactNode;
+  /** Icon content when isInProgress is set to true. Defaults to a medium spinner with 1em diameter.  */
+  progressIcon?: React.ReactNode;
+  /** Additional classes applied to the icon container */
+  className?: string;
+  /** Size of the icon component container and icon. */
+  size?: 'sm' | 'md' | 'lg' | 'xl';
+  /** Size of icon content. Overrides the icon size set by the size property. */
+  iconSize?: 'sm' | 'md' | 'lg' | 'xl';
```

Comment:
```suggestion
  iconSize?: 'sm' | 'md' | 'lg' | 'xl';
  /** Size of icon progress **/
  iconProgressSize?: 'sm' | 'md' | 'lg' | 'xl';
```

`iconProgress` also takes a size, just like `iconContent`
 https://github.com/patternfly/patternfly/blob/02bda75f8f22a9e230be216ed2e6e8ffd5b48cc3/src/patternfly/components/Icon/icon.scss#L89-L92

fwiw, they can both be set to a custom via an inline `--pf-c-icon__content--FontSize` though I don't know that we need to support that now.